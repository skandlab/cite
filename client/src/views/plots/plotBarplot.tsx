import React from "react";
import { echartsProps, barplotProps } from "./plotsConfig.json";
import { PlotEchartsBasic } from "./plotEchartsBasic";

interface Props {
	title?: string;
	xAxisName: string;
	yAxisName: string;
	barPlotData: {
		name: string;
		value: number;
	}[];
}

const mergeProps = (props: Props) => {
	return {
		...echartsProps,
		xAxis: {
			...echartsProps.xAxis,
			type: "category",
			data: props.barPlotData.map((d) => d.name),
			name: props.xAxisName,
			splitLine: {
				show: false,
			},
		},
		yAxis: {
			...echartsProps.yAxis,
			name: props.yAxisName,
		},
		series: {
			...barplotProps.series,
			data: props.barPlotData,
			tooltip: {
				formatter: function (params: any) {
					return params.name + "<br/>expression: " + params.value;
				},
			},
		},
	};
};

export const PlotBar = (props: Props) => (
	<PlotEchartsBasic option={mergeProps(props)} />
);
