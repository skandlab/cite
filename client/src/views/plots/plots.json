{
	"heatmapProps": {
		"indexBy": "tumorType",
		"minValue": 0,
		"maxValue": 100,
		"colors": "reds",
		"padding": 3,
		"margin": { "top": 24, "right": 0, "bottom": 0, "left": 48 },
		"axisTop": {
			"orient": "top",
			"tickSize": 5,
			"tickPadding": 5,
			"tickRotation": 0,
			"legend": "",
			"legendOffset": 36
		},
		"cellOpacity": 0.6,
		"enableLabels": false,
		"animate": false,
		"hoverTarget": "cell",
		"cellHoverOpacity": 0.6,
		"cellHoverOthersOpacity": 0.25
	}
}
