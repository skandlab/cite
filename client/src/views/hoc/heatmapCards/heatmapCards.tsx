import React from "react";
import styled from "@emotion/styled";
import { Card, Container } from "semantic-ui-react";
import { HeatMapCard, HeatmapCardProps } from "./heatmap";

interface HeatmapCardGroupProps {
	scoreDataList: HeatmapCardProps[];
}

export const HeatMapCardGroup = ({ scoreDataList }: HeatmapCardGroupProps) =>
	scoreDataList.length === 0 ? (
		<StyledContainer textAlign="center">
			<em>No Ligand-Receptor combination found.</em>
		</StyledContainer>
	) : (
		<StyledCardGroup data-tour="heatmapcards" centered doubling>
			{scoreDataList.map((scoreData, index) => (
				<HeatMapCard key={index} {...scoreData} />
			))}
		</StyledCardGroup>
	);

const StyledCardGroup = styled(Card.Group)`
	justify-content: space-around !important;
`;

const StyledContainer = styled(Container)`
	font-size: 20px;
	height: 30vh;
`;
