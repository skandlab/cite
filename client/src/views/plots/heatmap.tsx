import React from "react";
import { ResponsiveHeatMap } from "@nivo/heatmap";
import { heatmapProps } from "../plots/plots.json";

interface Props {
	data: { tumorType: string; [key: string]: string }[];
	pairKeys: string[];
	tumorKeys: string[];
	heatmapOptions?: {
		[key: string]: any;
	};
}

export const HeatMap = (props: Props) => (
	// @ts-ignore
	<ResponsiveHeatMap
		data={props.data.filter((prop) =>
			props.tumorKeys.includes(prop.tumorType)
		)}
		keys={props.pairKeys}
		{...{ ...heatmapProps, ...props.heatmapOptions }}
	/>
);
