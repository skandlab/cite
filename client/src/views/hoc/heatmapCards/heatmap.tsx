import React from "react";
import styled from "@emotion/styled";
import { Card, Statistic } from "semantic-ui-react";
import { PlotHeatMap } from "../../plots/plotHeatmap";
import { ROUTES } from "../../../utils/routes";
import { HeatmapCardType } from "../../../utils/interfaces";

export interface HeatmapCardProps extends HeatmapCardType {}

export const HeatMapCard = ({ ligand, receptor, data }: HeatmapCardProps) => (
	<StyledCard data-testid="displayCard">
		<Card.Content>
			<Card.Header>
				<Statistic.Group widths={2}>
					<Statistic className="xs" label="Ligand" value={ligand} />
					<Statistic
						className="xs"
						label="Receptor"
						value={receptor}
					/>
				</Statistic.Group>
			</Card.Header>
			<StyledCardDescription>
				<PlotHeatMap
					data={data}
					onHeatMapClick={(yKey: string) =>
						window.open(
							ROUTES.Expression.push(ligand, receptor, yKey),
							"_blank"
						)
					}
				/>
			</StyledCardDescription>
		</Card.Content>
	</StyledCard>
);

const StyledCard = styled(Card)`
	margin: 3em 0.8em !important;
`;

const StyledCardDescription = styled(Card.Description)`
	height: 400px !important;
`;
