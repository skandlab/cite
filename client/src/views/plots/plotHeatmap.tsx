import React from "react";
import { ResponsiveHeatMap } from "@nivo/heatmap";
import { heatmapProps } from "../plots/plotsConfig.json";

interface Props {
	data: { tumorType: string; [key: string]: string }[];
	xAxisFilterKeys: string[];
	yAxisFilterKeys: string[];
	heatmapOptions?: {
		[key: string]: any;
	};
	onHeatMapClick: (yKey: string) => void;
}

export const PlotHeatMap = (props: Props) => (
	// @ts-ignore
	<ResponsiveHeatMap
		data={props.data.filter((prop) =>
			props.yAxisFilterKeys.includes(prop.tumorType)
		)}
		keys={props.xAxisFilterKeys}
		onClick={(e) => props.onHeatMapClick(e.yKey as string)}
		{...{ ...heatmapProps, ...props.heatmapOptions }}
	/>
);
