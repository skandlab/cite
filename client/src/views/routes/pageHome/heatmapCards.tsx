import React from "react";
import { Card, Statistic } from "semantic-ui-react";
import { InterfaceData } from ".";
import { HeatMap } from "../../plots/heatmap";
import { ColumnBrowserProps } from "./columnBrowser/columnBrowser";

interface Props {
	data: InterfaceData[];
	pairsOptions: ColumnBrowserProps[];
	tumorOptions: ColumnBrowserProps[];
}

export const HeatMapCards = (props: Props) => {
	let _pairKeys = props.pairsOptions
		.filter((pair) => pair.isChecked)
		.map((pair) => pair.value);
	let _tumorKeys = props.tumorOptions
		.filter((tumor) => tumor.isChecked)
		.map((tumor) => tumor.value);

	if (_pairKeys.length === 0) {
		_pairKeys = props.pairsOptions.map((pair) => pair.value);
	}
	if (_tumorKeys.length === 0) {
		_tumorKeys = props.tumorOptions.map((tumor) => tumor.value);
	}

	return (
		<Card.Group centered>
			{props.data.map((_data, index) => (
				<Card className="heatmapcard" key={index}>
					<Card.Content>
						<Card.Header>
							<Statistic.Group widths={2}>
								<Statistic className="extra" size="mini" text>
									<Statistic.Value>
										{_data.ligand}
									</Statistic.Value>
									<Statistic.Label>Ligand</Statistic.Label>
								</Statistic>
								<Statistic className="extra" size="mini" text>
									<Statistic.Value>
										{_data.receptor}
									</Statistic.Value>
									<Statistic.Label>Receptor</Statistic.Label>
								</Statistic>
							</Statistic.Group>
						</Card.Header>
						<Card.Description style={{ height: "320px" }}>
							<HeatMap
								data={_data.scoreMatrix}
								pairKeys={_pairKeys}
								tumorKeys={_tumorKeys}
							/>
						</Card.Description>
					</Card.Content>
				</Card>
			))}
		</Card.Group>
	);
};
