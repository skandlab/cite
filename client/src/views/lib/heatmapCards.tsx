import React from "react";
import styled from "@emotion/styled";

import { Card, Statistic, Container } from "semantic-ui-react";
import { PlotHeatMap } from "../plots/plotHeatmap";

import { InterfaceScores } from "../../utils/interfaces";
import { ROUTES } from "../../utils/routes";

interface HeatMapCardsProps {
	xAxisFilterKeys: string[];
	yAxisFilterKeys: string[];
	data: InterfaceScores[];
}

export const HeatMapCards = (props: HeatMapCardsProps) => {
	const { data, ...restProps } = props;
	return data.length !== 0 ? (
		<StyledHeatmapCardGroup centered doubling>
			{data.map((_data, index) => (
				<HeatMapCard
					key={index}
					index={index}
					data={_data}
					{...restProps}
				/>
			))}
		</StyledHeatmapCardGroup>
	) : (
		<StyledContainer textAlign="center">
			<em>No Ligand-Receptor combination found.</em>
		</StyledContainer>
	);
};

interface HeatMapCardProps {
	data: InterfaceScores;
	index: number;
	xAxisFilterKeys: string[];
	yAxisFilterKeys: string[];
}

const HeatMapCard = ({ index, data, ...restProps }: HeatMapCardProps) => {
	const onHeatMapClick = (yKey: string) =>
		window.open(ROUTES.Expression.push(ligand, receptor, yKey), "_blank");

	const { ligand, receptor, scoreMatrix } = data;

	return (
		<StyledHeatmapCard key={index} data-testid="displayCard">
			<Card.Content>
				<Card.Header>
					<Statistic.Group widths={2}>
						<Statistic
							className="xs"
							label="Ligand"
							value={ligand}
						/>
						<Statistic
							className="xs"
							label="Receptor"
							value={receptor}
						/>
					</Statistic.Group>
				</Card.Header>
				<StyledHeatmapCardDescriptor>
					<PlotHeatMap
						{...restProps}
						data={scoreMatrix}
						onHeatMapClick={onHeatMapClick}
					/>
				</StyledHeatmapCardDescriptor>
			</Card.Content>
		</StyledHeatmapCard>
	);
};

const StyledHeatmapCardGroup = styled(Card.Group)`
	justify-content: space-around !important;
`;

const StyledHeatmapCard = styled(Card)`
	margin: 3em 0.8em !important;
`;

const StyledHeatmapCardDescriptor = styled(Card.Description)`
	height: 400px !important;
`;

const StyledContainer = styled(Container)`
	font-size: 20px;
	height: 30vh;
`;
