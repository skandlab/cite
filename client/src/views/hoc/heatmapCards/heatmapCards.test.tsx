import React from "react";
import "@testing-library/dom";
import "@testing-library/jest-dom/extend-expect";
import { screen } from "@testing-library/dom";
import { render, cleanup } from "@testing-library/react";
import { HeatMapCardGroup } from "./heatmapCards";
import { HeatMapDataType } from "../../plots/plotHeatmap";

const scoreData = {
	xyValues: [
		{ tumorType: "BRCA", cc: "1.08", cs: "0.22", nn: "8.33" },
		{ tumorType: "CRC", cc: "1.08", cs: "0.22", nn: "8.33" },
		{ tumorType: "LUAD", cc: "1.08", cs: "0.22", nn: "8.33" },
		{ tumorType: "SKCM", cc: "1.08", cs: "0.22", nn: "8.33" },
	],
	keys: ["cc", "cs", "nn"],
	indexBy: "tumorType",
};

const data: {
	ligand: string;
	receptor: string;
	data: HeatMapDataType;
}[] = [
	{ ligand: "BCAM", receptor: "NOX1", data: scoreData },
	{ ligand: "EGFR", receptor: "NOX1", data: scoreData },
	{ ligand: "BCAM", receptor: "KRAS", data: scoreData },
	{ ligand: "EGFR", receptor: "KRAS", data: scoreData },
];

const defaultRender = () => render(<HeatMapCardGroup scoreDataList={data} />);

const datalessRender = () => render(<HeatMapCardGroup scoreDataList={[]} />);

afterEach(() => {
	cleanup();
});

/**
 * cant render svgs
 * https://github.com/jsdom/jsdom/issues/300
 */
describe("render tests", () => {
	test("with data", () => {
		const { asFragment } = defaultRender();
		expect(asFragment()).toMatchSnapshot();
	});

	test("without data", () => {
		const { asFragment } = datalessRender();
		expect(asFragment()).toMatchSnapshot();

		screen.getByText("No Ligand-Receptor combination found.");
	});
});
