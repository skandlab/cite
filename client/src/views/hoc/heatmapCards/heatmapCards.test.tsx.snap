// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render tests with data 1`] = `
<DocumentFragment>
  <div
    class="ui centered doubling cards css-1iidtr2"
    data-tour="heatmapcards"
  >
    <div
      class="ui card css-1q0psf9"
      data-testid="displayCard"
    >
      <div
        class="content"
      >
        <div
          class="header"
        >
          <div
            class="ui two statistics"
          >
            <div
              class="ui statistic xs"
            >
              <div
                class="value"
              >
                BCAM
              </div>
              <div
                class="label"
              >
                Ligand
              </div>
            </div>
            <div
              class="ui statistic xs"
            >
              <div
                class="value"
              >
                NOX1
              </div>
              <div
                class="label"
              >
                Receptor
              </div>
            </div>
          </div>
        </div>
        <div
          class="description css-1vhobuo"
        >
          <div
            style="width: 100%; height: 100%;"
          />
        </div>
      </div>
    </div>
    <div
      class="ui card css-1q0psf9"
      data-testid="displayCard"
    >
      <div
        class="content"
      >
        <div
          class="header"
        >
          <div
            class="ui two statistics"
          >
            <div
              class="ui statistic xs"
            >
              <div
                class="value"
              >
                EGFR
              </div>
              <div
                class="label"
              >
                Ligand
              </div>
            </div>
            <div
              class="ui statistic xs"
            >
              <div
                class="value"
              >
                NOX1
              </div>
              <div
                class="label"
              >
                Receptor
              </div>
            </div>
          </div>
        </div>
        <div
          class="description css-1vhobuo"
        >
          <div
            style="width: 100%; height: 100%;"
          />
        </div>
      </div>
    </div>
    <div
      class="ui card css-1q0psf9"
      data-testid="displayCard"
    >
      <div
        class="content"
      >
        <div
          class="header"
        >
          <div
            class="ui two statistics"
          >
            <div
              class="ui statistic xs"
            >
              <div
                class="value"
              >
                BCAM
              </div>
              <div
                class="label"
              >
                Ligand
              </div>
            </div>
            <div
              class="ui statistic xs"
            >
              <div
                class="value"
              >
                KRAS
              </div>
              <div
                class="label"
              >
                Receptor
              </div>
            </div>
          </div>
        </div>
        <div
          class="description css-1vhobuo"
        >
          <div
            style="width: 100%; height: 100%;"
          />
        </div>
      </div>
    </div>
    <div
      class="ui card css-1q0psf9"
      data-testid="displayCard"
    >
      <div
        class="content"
      >
        <div
          class="header"
        >
          <div
            class="ui two statistics"
          >
            <div
              class="ui statistic xs"
            >
              <div
                class="value"
              >
                EGFR
              </div>
              <div
                class="label"
              >
                Ligand
              </div>
            </div>
            <div
              class="ui statistic xs"
            >
              <div
                class="value"
              >
                KRAS
              </div>
              <div
                class="label"
              >
                Receptor
              </div>
            </div>
          </div>
        </div>
        <div
          class="description css-1vhobuo"
        >
          <div
            style="width: 100%; height: 100%;"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`render tests without data 1`] = `
<DocumentFragment>
  <div
    class="ui center aligned container css-khn4ns"
  >
    <em>
      No Ligand-Receptor combination found.
    </em>
  </div>
</DocumentFragment>
`;
